examples/native-fungible/src/contract.rs

```diff
@@ -0,0 +1,287 @@
+            }
+
+            ApplicationCall::Transfer {
+                owner,
+                amount,
+                destination,
+            } => {
+                Self::check_account_authentication(context.authenticated_signer, owner)?;
+                let owner = self.normalize_owner(owner);
+                let target_account = self.destination_to_account(destination);
+
+                system_api::transfer(Some(owner), target_account, amount);
+                Ok(ApplicationCallOutcome::default())
+            }
+
+            ApplicationCall::Claim {
+                source_account,
+                amount,
+                target_account,
+            } => {
+                Self::check_account_authentication(
+                    context.authenticated_signer,
+                    source_account.owner,
+                )?;
+
+                let source_account = self.normalize_account(source_account);
+                let target_account = self.normalize_account(target_account);
+
+                system_api::claim(source_account, target_account, amount);
+                Ok(ApplicationCallOutcome::default())
+            }
+
+            ApplicationCall::TickerSymbol => {
+                let outcome = ApplicationCallOutcome {
+                    value: FungibleResponse::TickerSymbol(String::from(TICKER_SYMBOL)),
+                    ..Default::default()
+                };
+                Ok(outcome)
+            }
+        }
+    }
+
+    async fn handle_session_call(
+        &mut self,
+        _context: &CalleeContext,
+        _state: Self::SessionState,
+        _request: Self::SessionCall,
+        _forwarded_sessions: Vec<SessionId>,
+    ) -> Result<SessionCallOutcome<Self::Message, Self::Response, Self::SessionState>, Self::Error>
+    {
+        Err(Error::SessionsNotSupported)
+    }
+}
+
+impl NativeFungibleToken {
+    fn get_transfer_outcome(
+        &self,
+        source: AccountOwner,
+        target_account: fungible::Account,
+        amount: Amount,
+    ) -> ExecutionOutcome<Message> {
+        if target_account.chain_id == system_api::current_chain_id() {
+            ExecutionOutcome::default()
+        } else {
+            let message = Message::Credit {
+                target: target_account.owner,
+                amount,
+                source,
+            };
+
+            ExecutionOutcome::default().with_tracked_message(target_account.chain_id, message)
+        }
+    }
+
+    fn normalize_owner(&self, account_owner: AccountOwner) -> Owner {
+        match account_owner {
+            AccountOwner::User(owner) => owner,
+            AccountOwner::Application(_) => panic!("Applications not supported yet!"),
+        }
+    }
+
+    fn normalize_account(&self, account: fungible::Account) -> Account {
+        let owner = self.normalize_owner(account.owner);
+        Account {
+            chain_id: account.chain_id,
+            owner: Some(owner),
+        }
+    }
```

Comment:
Maybe we should (in `fungible` itself) `impl TryFrom<fungible::Account> for linera_sdk::base::Account`, and `impl From<linera_sdk::base::Account> for fungible::Account`?